minidump-analyzer/src/main.rs

```diff
@@ -0,0 +1,363 @@
+                                std::mem::transmute::<
+                                    &MinidumpThread,
+                                    &'static MinidumpThread<'static>,
+                                >(thread)
+                            };
+                            tokio::spawn(async move {
+                                context.thread_call_stack(thread_idx, thread).await
+                            })
+                        }),
+                    )
+                    .await
+                })
+                .into_iter()
+                .collect::<Result<_, _>>()
+                .context("while spawning tasks")
+        }
+
+        /// Get all modules, ordered by address.
+        pub fn modules(&self) -> impl Iterator<Item = &MinidumpModule> {
+            self.context.module_list.by_addr()
+        }
+
+        /// Get the index of the given module in the iterator returned by `modules`.
+        pub fn module_index(&self, module: &MinidumpModule) -> Option<usize> {
+            self.modules()
+                .position(|o| module.base_address() == o.base_address())
+        }
+
+        /// Get the index of the main module.
+        ///
+        /// Returns `None` when no main module exists (only when there are modules).
+        pub fn main_module_index(&self) -> Option<usize> {
+            self.context
+                .module_list
+                .main_module()
+                .and_then(|m| self.module_index(m))
+        }
+
+        /// Convert a call stack to json (in a form appropriate for the extra json file).
+        pub fn call_stack_to_json(&self, call_stack: &CallStack) -> json::JsonValue {
+            json::object! {
+                "frames": call_stack.frames.iter().map(|frame| {
+                    json::object! {
+                        "ip": frame.instruction,
+                        "module_index": frame.module.as_ref().and_then(|m| self.module_index(m)),
+                        "trust": match frame.trust {
+                            FrameTrust::Context => "context",
+                            FrameTrust::PreWalked => "prewalked",
+                            FrameTrust::CallFrameInfo => "cfi",
+                            FrameTrust::FramePointer => "frame_pointer",
+                            FrameTrust::CfiScan => "cfi_scan",
+                            FrameTrust::Scan => "scan",
+                            FrameTrust::None => "none"
+                        }
+                    }
+                }).collect::<Vec<_>>()
+            }
+        }
+
+        /// Get the json representation of module signature information.
+        pub fn module_signature_info(&self) -> json::JsonValue {
+            // TODO { binary_org_name: [code_file filename] }
+            // authenticode_parser uses system openssl, which may be undesirable. In that case, we may
+            // want to directly use `winapi`.
+            json::JsonValue::Null
```

Comment:
I'd go for winapi here for the time being, for the sake of parity. We can modify this once more mature crates emerge at a later point.